iothub/device/src/ClientOptions.cs

```diff
@@ -54,5 +54,11 @@ public class ClientOptions
         /// or the <see cref="ModuleClient.CreateFromEnvironmentAsync(ClientOptions)"/> flow.
         /// </remarks>
         public int SasTokenRenewalBuffer { get; set; }
+
+        /// <summary>
+        /// A keep-alive for when using web sockets.
+        /// </summary>
+        /// <seealso href="https://docs.microsoft.com/dotnet/api/system.net.websockets.clientwebsocketoptions.keepaliveinterval"/>
+        public TimeSpan? WebSocketKeepAlive { get; set; }
```

Comment:
I like this as a home for this new option. I don't know if this has been discussed yet, but I'd like to see us remove the transport specific settings classes when we move to v2. With that in mind, I like adding to ClientOptions like this.

Reply:
I too like it here since it impacts both transports.

Reply:
I'm probably not a fan of consolidating everything, or at least having transport-specific sections in `ClientOptions`. We have enough of a problem today with this strategy pattern where not everything applies to every transport. I think we're better off keeping things separate to avoid confusion.